src/components/SearchMap.js

```diff
@@ -107,12 +107,20 @@ const SearchMap = ({mapLoaded, lastKnownRegion, updateRegion, sites, navigate, s
                     <TouchableOpacity style={touchableMainContainerStyle} onPress={() => getSiteDetail({selectedSite: selectedSite, navigate})}>
                         <View style={mainInnerContainerStyle}>
                             <View style={topRowInfoStyle}>
-                                <View>
+                                <Icon style={closeIconStyle}
+                                      type='ionicon'
+                                      name='md-close-circle'
+                                      size={40}
+                                      color={'white'}
+                                      onPress={() => getSiteDetail({selectedSite: null})}
+                                />
 
-                                </View>
-                                <Icon style={closeIconStyle} type='ionicon' name='md-close-circle' size={40}
+                                <Icon type='ionicon'
+                                      name={isFavorite ? 'ios-heart' : 'ios-heart-outline'}
+                                      size={40}
                                       color={'white'}
-                                      onPress={() => getSiteDetail({selectedSite: null})}/>
+                                      onPress={toggleSiteFavorite}
```

Comment:
This ended up being pretty straightforward... just brought the same logic we were using in `SiteDetailScreen` to add Favorites functionality to the MapView site preview as well. This diff looks confusing because I also swapped the locations so that the "close" button is now on the top right, and the "Favorites" button is on the top right of the preview. This is more in line with how we navigate the rest of the app. The "Back" button is always on the top left, and the "Favorites" icon is on the top right of the `SiteDetailScreen`